### ISSUE TITLE
S2327 triggers even when there is code in between the two blocks

### ISSUE THREAD

**Issue by YairHalberstadt** [SonarSource/sonar-dotnet#2490]
### Description

S2327: "try" statements with identical "catch" and/or "finally" blocks should be merged

Should not trigger if there is code in between the two blocks, as it is likely you don't want to handle exceptions from that code. Eg:

```csharp
			try
			{
				await _dataAccessService.AddAsync(history).ConfigureAwait(false);
			}
			catch (Exception e)
			{
				_logger.Error($"failed to save history to database : {e}");
			}

			var status = await HandleMessageWithProxies(context, request, subscriptionCancelledToken).ConfigureAwait(false);

			try
			{
				history.Completed = Timestamp.FromDateTime(DateTime.UtcNow);
				history.Outcome = GetOutcome();
				history.Error = status.Error;

				await _dataAccessService.AddAsync(history).ConfigureAwait(false);
			}
			catch (Exception e)
			{
				_logger.Error($"failed to save history to database : {e}");
			}
```

### Expected behavior

no warning

### Actual behavior

warning S2327 // "try" statements with identical "catch" and/or "finally" blocks should be merged

### Related information

* SonarC# Version 7.14.0.8411

* Visual Studio Version 16.1.3

**Comment by scharnyw** [comment]
IMHO this rule is fundamentally flawed. The suggested "solution" almost always changes the semantics of the program and in general it leads to much more problems than it solves.

**Comment by andrei-epure-sonarsource** [comment]
(this rule is not in the default quality profile anymore for C#)

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1987](https://sonarsource.atlassian.net/browse/NET-1987)

[NET-1987]: https://sonarsource.atlassian.net/browse/NET-1987?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
